### PR TITLE
Base functionality for code driven documentation website

### PR DESCRIPTION
**Pull request by bilke**
I apologize for this large PR! 😜 

This introduce the possibility to have the documentation version controlled together with the code itself. Eventually this will make docs.opengeosys.org and maybe more .. obsolete. 

## The big picture

- Development related content such as developer guide, benchmark documentation, tools description, ... will be simple Markdown files in e.g. [/web/content/docs](https://github.com/bilke/ogs/blob/web-hugo/web/content/docs/benchmarks/elliptic/groundwater-flow-neumann.md)
- You can preview documentation locally with [Hugo](https://gohugo.io) – a static site generator
- Other content such as news, blog posts, articles can be authored in a graphical environment and is not part of the code but hosted with a content management system (currently a prototypical implementation uses [Contenful](https://www.contentful.com/) but maybe we can even stick to the current system [Craft CMS](https://craftcms.com/) once version 3 is released). This will lower barriers for non-technical people to contribute to the site. The content gets imported automatically on Jenkins or locally (see [web/README.md](https://github.com/bilke/ogs/blob/web-hugo/web/README.md) for details)
- You can [mark](https://github.com/bilke/ogs/blob/web-hugo/ProcessLib/GroundwaterFlow/CMakeLists.txt#L80) benchmarks to be automatically and interactively visualized [in the documentation](https://github.com/bilke/ogs/commit/d4fc7d94a3821a6b4483a1d7aeaabd6ee391c449#diff-2f5b1ac2a759aa09b2d3f5cc1ece45ceR108) inside your [browser](https://dev.opengeosys.org/docs/benchmarks/elliptic/groundwater-flow-neumann/#results-and-evaluation) via [vtk.js](https://kitware.github.io/vtk-js/)! 🍻 

### Notable changes

- [git-lfs](https://git-lfs.github.com/) is used for image files in the documentation section, so this is a new requirement if you want to edit the docs
- New build target `web` can be used to build the docs
- Site is built by Jenkins on every build (also on PRs), just click in the link called [Web](https://jenkins.opengeosys.org/job/User/job/bilke/job/ogs/job/web-hugo/Web/) on the sidebar
- Site built from `ufz/master` is served at https://dev.opengeosys.org at the moment

### Requirements

All of the following should be optional if you don't want to deal with the site.

- [Hugo](https://gohugo.io), [NodeJS](https://nodejs.org/en/) for web site generation
- Python, pip and super secret credentials for importing content from CMS
- git-lfs to checkout images
- ParaView installed for converting benchmark result files to vtk-js for web visualization

## Next steps

- [ ] Merge this PR!

Next next steps:

- [x] Fix docs index pages and navigation
- [ ] Migrate all content from the [Selected benchmarks](https://docs.opengeosys.org/docs/benchmarks) section to this structure
- [ ] Remove this section from docs.opengeosys.org and instead link to here
- [ ] Migrate developer guide
- [ ] Create a developer guide page about this
- [ ] Migrate tools
- [ ] Refine CMS related stuff & get content in
- [ ] Refine visualization functionality
- [ ] Style the site
- [ ] 🕺

### CODE REVIEW

File: web/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+## Getting started
+
+- Download [Hugo](https://gohugo.io/#action) and put it in your `PATH`
+- Install Node packages with `npm install` (or via `yarn`), *OPTIONAL* for SCSS and Javascript
```
Comment (endJunction) [inline web/README.md]: Please add info where to run the `npm` and `pip`. Source dir or build dir?

File: web/content/about.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
++++
+title = "About"
+author = "Lars Bilke"
+draft = true
+date = "2017-01-13T14:24:23+01:00"
+
++++
+
+Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```
Comment (endJunction) [inline web/content/about.md]: Copy the readme content...

File: web/import/secret_example.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ogsSpace = '4nuqzxntzxks'
```
Comment (endJunction) [inline web/import/secret_example.py]: This is not a key to something, isn't it?

File: web/README.md (hunk 1 above)
Comment (endJunction) [inline web/README.md]: Below there is a yarn.lock file. If yarn is optional, how is it updated? --> for discussion.

**Comment by endJunction** [comment]
The web/LICENSE needs some content.

File: web/README.md (hunk 1 above)
Comment (bilke) [inline web/README.md]: This file is updated if you run `yarn upgrade`. I will do it when it is necessary (e.g. to get a new version of vtk.js).

File: web/import/secret_example.py (hunk 3 above)
Comment (bilke) [inline web/import/secret_example.py]: No this is an ID. You need the access token to use the API.

**Comment by bilke** [comment]
@endJunction What do you mean with web/LICENSE? Should the content of the web-folder have a different license?

File: web/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+- Install [Node.js](https://nodejs.org/en/), check if `npm` is in your `PATH`
+- Install [Yarn](https://yarnpkg.com/en/docs/install), *OPTIONAL* faster alternative to `npm install`
+- Install Python and `pip`, *OPTIONAL* for getting content from [Contentful](https://contentful.com)
+- Install [ParaView](http://www.paraview.org/download/), *OPTIONAL* for converting VTK output files to vtk.js-format for interactive web visualization, check if `pvpython` is either in the `PATH` or ParaView is installed in `/Applications/` on macOS or `/usr/local/opt/paraview` on Linux
+
+## Getting started
+
+- Inside the source-directory `ogs/web`:
+  - Install Node packages with `npm install` (or via `yarn`)
+  - Install Python packages with `pip install -r requirements.txt`, *OPTIONAL* for getting content from [Contentful](https://contentful.com)
```
Comment (endJunction) [inline web/README.md]: Got an error executing pip. Not sure what the reasons are:
```
➜  web git:(8d03ba34d) pip install -r requirements.txt
Collecting requests==2.12.4 (from -r requirements.txt (line 1))
  Downloading requests-2.12.4-py2.py3-none-any.whl (576kB)
    100% |████████████████████████████████| 583kB 1.8MB/s 
Installing collected packages: requests
  Found existing installation: requests 2.13.0
    Uninstalling requests-2.13.0:
Exception:
Traceback (most recent call last):
  File "/usr/lib/python3.6/shutil.py", line 538, in move
    os.rename(src, real_dst)
OSError: [Errno 18] Invalid cross-device link: '/usr/lib/python3.6/site-packages/requests' -> '/tmp/pip-l6931pqo-uninstall/usr/lib/python3.6/site-packages/requests'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/usr/lib/python3.6/site-packages/pip/req/req_set.py", line 778, in install
    requirement.uninstall(auto_confirm=True)
  File "/usr/lib/python3.6/site-packages/pip/req/req_install.py", line 754, in uninstall
    paths_to_remove.remove(auto_confirm)
  File "/usr/lib/python3.6/site-packages/pip/req/req_uninstall.py", line 115, in remove
    renames(path, new_path)
  File "/usr/lib/python3.6/site-packages/pip/utils/__init__.py", line 267, in renames
    shutil.move(old, new)
  File "/usr/lib/python3.6/shutil.py", line 550, in move
    rmtree(src)
  File "/usr/lib/python3.6/shutil.py", line 474, in rmtree
    _rmtree_safe_fd(fd, path, onerror)
  File "/usr/lib/python3.6/shutil.py", line 412, in _rmtree_safe_fd
    _rmtree_safe_fd(dirfd, fullname, onerror)
  File "/usr/lib/python3.6/shutil.py", line 432, in _rmtree_safe_fd
    onerror(os.unlink, fullname, sys.exc_info())
  File "/usr/lib/python3.6/shutil.py", line 430, in _rmtree_safe_fd
    os.unlink(name, dir_fd=topfd)
PermissionError: [Errno 13] Permission denied: 'certs.cpython-36.opt-1.pyc'
```

Comment (bilke) [inline web/README.md]: maybe `sudo pip ...`?

**Pull request merged**